src/java.base/unix/classes/sun/nio/fs/UnixPath.java

```diff
@@ -90,8 +90,10 @@ static String normalizeAndCheck(String input) {
             checkNotNul(input, c);
             prevChar = c;
         }
-        if (prevChar == '/')
-            return normalize(input, n, n - 1);
+        if (prevChar == '/') {
+            if (n == 1) return "/";
+            return input.substring(0, n - 1);
+        }
```

Comment:
I think the changes in this PR are okay as they don't impact the common case. Style wise I think I'd prefer if L95-L95 were changed to `return (n == 1) ? "/"  : input.substring(0, n - 1);`